src/ensemble/random_forest_classifier.rs

```diff
@@ -316,6 +317,36 @@ impl<T: RealNumber> RandomForestClassifier<T> {
         which_max(&result)
     }
 
+    /// Predict the per-class probabilties for each observation. The probability is calculated as the fraction of trees that predicted a given class
+    pub fn predict_probs<M: Matrix<T>>(&self, x: &M) -> Result<DenseMatrix<f64>, Failed> {
```

Comment:
Could you please cover this method, as well as the function `predict_probs_for_row` with unit test? This will help to get your code through automatic checks